File: examples/example-kubernetes-client-openapi-watcher-java/src/main/java/micronaut/client/SecretController.java
package micronaut.client;

import io.micronaut.http.annotation.Controller;
import io.micronaut.http.annotation.Get;
import io.micronaut.kubernetes.client.openapi.watcher.api.CoreV1ApiWatcher;
import jakarta.inject.Inject;
import jakarta.validation.constraints.NotNull;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

@Controller("/secrets")
public class SecretController {
    private static final Logger LOG = LoggerFactory.getLogger(SecretController.class);

    @Inject
    CoreV1ApiWatcher coreV1ApiWatcher;

    @Get("/{namespace}")
    public void startWatchingSecrets(final @NotNull String namespace) {
        coreV1ApiWatcher.listNamespacedSecret(namespace, null, null, null, null, null, null, null, null, null, null, null, true)
            .subscribe(event -> LOG.info(event.toString()));
    }
}

File: examples/example-kubernetes-client-openapi-watcher-kotlin/src/main/kotlin/micronaut/client/SecretController.kt
package micronaut.client

import io.micronaut.http.annotation.Controller
import io.micronaut.http.annotation.Get
import io.micronaut.kubernetes.client.openapi.watcher.api.CoreV1ApiWatcher
import jakarta.inject.Inject
import jakarta.validation.constraints.NotNull
import org.slf4j.LoggerFactory

@Controller("/secrets")
open class SecretController {

    companion object {
        private val LOG = LoggerFactory.getLogger(SecretController::class.java)
    }

    @Inject
    lateinit var coreV1ApiWatcher: CoreV1ApiWatcher

    @Get("/{namespace}")
    open fun startWatchingSecrets(@NotNull namespace: String) {
        coreV1ApiWatcher.listNamespacedSecret(namespace, null, null, null, null, null, null, null, null, null, null, null, true)
            .subscribe { event -> LOG.info(event.toString()) }
    }
}
